services/_shared/adapters/pg_install_store.py

```
@@ -286,14 +295,34 @@ def set_repo_config(
             f"set_repo_config() persona flags must be bool or None; got {non_bool!r}"
         )
+    # Voice is an exception: it's a Literal["caveman", "sage"] (or None).
+    if "elder_voice" in persona_flags:
+        voice_val = persona_flags["elder_voice"]
+        if voice_val == "sage":
+            from adapters.pg_install_store import is_install_allowlisted
+            if not is_install_allowlisted(install_id):
+                raise ValueError(
+                    f"Elder sage voice requires an allowlisted (paid) installation. "
+                    f"Install {install_id} is not entitled."
+                )
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)